src/components/Equations.ts

```diff
@@ -106,6 +106,7 @@ export class Equations<TGameStartr extends FullScreenPokemon> extends Component<
      * @returns A newly created Pokemon.
      */
     public newPokemon(chosenInfo: INewPokemon): IPokemon {
+        this.gameStarter.modAttacher.fireEvent("onRandomItems", chosenInfo);
```

Comment:
Event triggers aren't what you _want to happen_, they are _what's happening_. This would be something about the new Pokemon instead of `"onRandomItems"`.